src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.Import.cs

```diff
@@ -35,9 +47,70 @@ public static CngKey Import(byte[] keyBlob, CngKeyBlobFormat format, CngProvider
 
             SafeNCryptProviderHandle providerHandle = provider.OpenStorageProvider();
             SafeNCryptKeyHandle keyHandle;
-            ErrorCode errorCode = Interop.NCrypt.NCryptImportKey(providerHandle, IntPtr.Zero, format.Format, IntPtr.Zero, out keyHandle, keyBlob, keyBlob.Length, 0);
-            if (errorCode != ErrorCode.ERROR_SUCCESS)
-                throw errorCode.ToCryptographicException();
+            ErrorCode errorCode;
+            
+            if (curve == null)
+            {
+                errorCode = Interop.NCrypt.NCryptImportKey(providerHandle, IntPtr.Zero, format.Format, IntPtr.Zero, out keyHandle, keyBlob, keyBlob.Length, 0);
+                if (errorCode != ErrorCode.ERROR_SUCCESS)
+                {
+                    Exception e = errorCode.ToCryptographicException();
+                    if (errorCode == ErrorCode.NTE_NOT_SUPPORTED &&
+                        (format == CngKeyBlobFormat.EccFullPrivateBlob || format == CngKeyBlobFormat.EccFullPrivateBlob))
+                    {
+                        // Attempt to import on < Windows 10
+                        throw new PlatformNotSupportedException(SR.Cryptography_InvalidCurveOid, e);
```

Comment:
Good catch. Wasn't an existing resource that matched perfectly, but for lack of creating a very similar resource changed to
throw new PlatformNotSupportedException(string.Format(SR.Cryptography_CurveNotSupported, format.Format), e);

where string is
    <value>The specified curve '{0}' or its parameters are not valid for this platform.</value>

so {0} will be "ECCFULLPRIVATEBLOB" for example